std/encoding.d

```diff
@@ -378,28 +378,28 @@ template EncoderFunctions()
     template ReadFromString()
     {
         @property bool canRead() { return s.length != 0; }
-        E peek() { return s[0]; }
-        E read() { E t = s[0]; s = s[1..$]; return t; }
+        E peek() @safe pure @nogc nothrow { return s[0]; }
+        E read() @safe pure @nogc nothrow { E t = s[0]; s = s[1..$]; return t; }
```

Comment:
A bit off-topic, but would it make sense to define a D-style order of the attributes? Everyone seems to have their own concept (or none) and even this PR contains multiple styles. How about something like
- first `@`s, then other attributes, both sorted alphabetically 
- all attributes should be sorted alphabetically 
-...